pilot/cmd/pilot-agent/main.go

```diff
@@ -141,6 +141,8 @@ var (
 	eccSigAlgEnv        = env.RegisterStringVar("ECC_SIGNATURE_ALGORITHM", "", "The type of ECC signature algorithm to use when generating private keys").Get()
 	fileMountedCertsEnv = env.RegisterBoolVar("FILE_MOUNTED_CERTS", false, "").Get()
 	useTokenForCSREnv   = env.RegisterBoolVar("USE_TOKEN_FOR_CSR", false, "CSR requires a token").Get()
+	credFetcherTypeEnv  = env.RegisterStringVar("CREDENTIAL_FETCHER_TYPE", "Kubernetes",
```

Comment:
I don't think 'kubernetes' credential fetcher makes any sense. We use a file - which is clean and simple.

We can have an optional plugin for other metadata server integration - or use one of those supported by grpc, but no need to over-complicate code that already works well.

Reply:
I prefer to keep the interface consistent. Otherwise, in every place GetPlatformCredential (or other functions in credfetcher) is called, I have to check "if k8s, read file. Otherwise, call getPlatformCredential". With the current way, it is simply a GetPlatformCredential, and you don't need to specify handle k8s case separately.

Reply:
I can also rename "Kubernetes" to "TokenFile" if you think that's more accurate.

Reply:
I think the agreement was for the PR that is backported to 1.7 to not touch the stable code using token file, you are adding a feature ( GCP VM ), no need to combine that with changes to other platforms. Just add the interface and the GCP implementation.

For master - the agreement was to expand the discussion to include gRPC people, and move ahead with a refactoring where we can reuse code that already exists. I have no problem to switch 'file based' token to a different implementation, but like the metadata server I prefer to share code with gRPC, which also is attempting to support Istio and XDS.